portability-core/src/test/java/org/dataportabilityproject/serviceProviders/google/calendar/GoogleCalendarServiceTest.java

```diff
@@ -21,72 +21,108 @@
 
   private static final String NEXT_TOKEN = "next_token";
 
   private GoogleCalendarService calendarService;
 
   private Calendar calendarClient;
   private JobDataCache jobDataCache;
+  private Calendar.Calendars calendarCalendars;
   private Calendar.CalendarList calendarCalendarList;
-  private Calendar.CalendarList.List listRequest;
-  private CalendarList listResponse;
+  private Calendar.CalendarList.List calendarListRequest;
+  private Calendar.Calendars.Insert calendarInsertRequest;
+  private CalendarList calendarListResponse;
+  private Calendar.Events calendarEvents;
+  private Calendar.Events.List eventListRequest;
+  private Calendar.Events.Insert eventInsertRequest;
+  private Events eventListResponse;
 
   @Before
   public void setup() throws IOException {
+    calendarCalendars = mock(Calendar.Calendars.class);
     calendarCalendarList = mock(Calendar.CalendarList.class);
-    listRequest = mock(Calendar.CalendarList.List.class);
+    calendarListRequest = mock(Calendar.CalendarList.List.class);
+    calendarInsertRequest = mock(Calendar.Calendars.Insert.class);
+    calendarEvents = mock(Calendar.Events.class);
+    eventListRequest = mock(Calendar.Events.List.class);
+    eventInsertRequest = mock(Calendar.Events.Insert.class);
 
     calendarClient = mock(Calendar.class);
-    jobDataCache = mock(JobDataCache.class);
+    jobDataCache = new InMemoryJobDataCache();
 
     calendarService = new GoogleCalendarService(calendarClient, jobDataCache);
 
+    when(calendarClient.calendars()).thenReturn(calendarCalendars);
+
     when(calendarClient.calendarList()).thenReturn(calendarCalendarList);
-    when(calendarCalendarList.list()).thenReturn(listRequest);
+    when(calendarCalendarList.list()).thenReturn(calendarListRequest);
+    when(calendarClient.events()).thenReturn(calendarEvents);
+
+    when(calendarEvents.list(CALENDAR_ID)).thenReturn(eventListRequest);
+    when(eventListRequest.setMaxAttendees(GoogleStaticObjects.MAX_ATTENDEES))
+        .thenReturn(eventListRequest);
   }
 
+  /**
+   * Sets up a response with a single calendar, containing a single event
+   */
   private void setUpSingleCalendarResponse() throws IOException {
-    listResponse = new CalendarList().setItems(Collections.singletonList(CALENDAR_LIST_ENTRY));
+    setUpSingleEventResponse();
+    calendarListResponse = new CalendarList()
+        .setItems(Collections.singletonList(CALENDAR_LIST_ENTRY));
+
+    when(calendarListRequest.execute()).thenReturn(calendarListResponse);
+  }
 
-    when(listRequest.execute()).thenReturn(listResponse);
+  /**
+   * Sets up a response for a single event
+   */
+  private void setUpSingleEventResponse() throws IOException {
+    eventListResponse = new Events().setItems(Collections.singletonList(EVENT));
+    when(eventListRequest.execute()).thenReturn(eventListResponse);
   }
 
   @Test
-  public void testExportFirstSet() throws IOException {
+  public void testExportCalendarFirstSet() throws IOException {
```

Comment:
Would First or Single be more accurate for this test?

Reply:
I think the fact that it's testing the export of the first calendar page is more important.